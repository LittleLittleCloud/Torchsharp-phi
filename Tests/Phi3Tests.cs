using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ApprovalTests;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static TorchSharp.torch;
using TorchSharp;
using Xunit;
using FluentAssertions;

namespace Phi.Tests;

public class Phi3Tests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("Approvals")]
    public async Task ShapeTest()
    {
        var modelWeightFolder = "C:\\Users\\xiaoyuz\\source\\repos\\Phi-3-mini-4k-instruct";
        var model = Phi3ForCasualLM.FromPretrained(modelWeightFolder, torchDtype: ScalarType.BFloat16);
        var state_dict_str = model.Peek();
        Approvals.Verify(state_dict_str);
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("Approvals")]
    public async Task TokenizerTest()
    {
        var modelWeightFolder = "C:\\Users\\xiaoyuz\\source\\repos\\Phi-3-mini-4k-instruct";
        var tokenizer = LLama2Tokenizer.FromPretrained(modelWeightFolder);
        tokenizer.BosId.Should().Be(1);
        tokenizer.EosId.Should().Be(32000);
        var messages = new string[]
        {
            "Can you provide ways to eat combinations of bananas and dragonfruits?",
            "Sure! Here are some ways to eat bananas and dragonfruits together: 1. Banana and dragonfruit smoothie: Blend bananas and dragonfruits together with some milk and honey. 2. Banana and dragonfruit salad: Mix sliced bananas and dragonfruits together with some lemon juice and honey.",
            "What about solving an 2x + 3 = 7 equation?"
        };
        var sb = new StringBuilder();
        foreach (var message in messages)
        {
            var tokenized = tokenizer.Encode(message, true, false);
            var tokenized_str = string.Join(", ", tokenized.Select(x => x.ToString()));

            sb.AppendLine(tokenized_str);
        }
        Approvals.Verify(sb.ToString());
    }
}
