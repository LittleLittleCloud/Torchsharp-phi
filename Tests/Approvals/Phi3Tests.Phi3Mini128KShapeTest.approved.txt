0: lm_head.weight: sum: 7120.249  dtype: BFloat16 shape: [32064,3072]
1: model.embed_tokens.weight: sum: 7057.233  dtype: BFloat16 shape: [32064,3072]
2: model.layers.0.input_layernorm.weight: sum: 36.7826  dtype: BFloat16 shape: [3072]
3: model.layers.0.mlp.down_proj.weight: sum: 3005.3293  dtype: BFloat16 shape: [3072,8192]
4: model.layers.0.mlp.gate_up_proj.weight: sum: 4135.2334  dtype: BFloat16 shape: [16384,3072]
5: model.layers.0.post_attention_layernorm.weight: sum: 36.9759  dtype: BFloat16 shape: [3072]
6: model.layers.0.self_attn.o_proj.weight: sum: 2271.7014  dtype: BFloat16 shape: [3072,3072]
7: model.layers.0.self_attn.qkv_proj.weight: sum: 2120.562  dtype: BFloat16 shape: [9216,3072]
8: model.layers.1.input_layernorm.weight: sum: 37.0373  dtype: BFloat16 shape: [3072]
9: model.layers.1.mlp.down_proj.weight: sum: -1839.5228  dtype: BFloat16 shape: [3072,8192]
10: model.layers.1.mlp.gate_up_proj.weight: sum: 4790.403  dtype: BFloat16 shape: [16384,3072]
11: model.layers.1.post_attention_layernorm.weight: sum: 37.1076  dtype: BFloat16 shape: [3072]
12: model.layers.1.self_attn.o_proj.weight: sum: -8700.337  dtype: BFloat16 shape: [3072,3072]
13: model.layers.1.self_attn.qkv_proj.weight: sum: 4062.7163  dtype: BFloat16 shape: [9216,3072]
14: model.layers.10.input_layernorm.weight: sum: 36.8355  dtype: BFloat16 shape: [3072]
15: model.layers.10.mlp.down_proj.weight: sum: 2782.592  dtype: BFloat16 shape: [3072,8192]
16: model.layers.10.mlp.gate_up_proj.weight: sum: 9903.472  dtype: BFloat16 shape: [16384,3072]
17: model.layers.10.post_attention_layernorm.weight: sum: 36.9139  dtype: BFloat16 shape: [3072]
18: model.layers.10.self_attn.o_proj.weight: sum: 1620.5906  dtype: BFloat16 shape: [3072,3072]
19: model.layers.10.self_attn.qkv_proj.weight: sum: 3885.4875  dtype: BFloat16 shape: [9216,3072]
20: model.layers.11.input_layernorm.weight: sum: 36.9525  dtype: BFloat16 shape: [3072]
21: model.layers.11.mlp.down_proj.weight: sum: 3103.3032  dtype: BFloat16 shape: [3072,8192]
22: model.layers.11.mlp.gate_up_proj.weight: sum: 1586.5453  dtype: BFloat16 shape: [16384,3072]
23: model.layers.11.post_attention_layernorm.weight: sum: 36.8569  dtype: BFloat16 shape: [3072]
24: model.layers.11.self_attn.o_proj.weight: sum: 2572.287  dtype: BFloat16 shape: [3072,3072]
25: model.layers.11.self_attn.qkv_proj.weight: sum: 4441.204  dtype: BFloat16 shape: [9216,3072]
26: model.layers.12.input_layernorm.weight: sum: 36.9514  dtype: BFloat16 shape: [3072]
27: model.layers.12.mlp.down_proj.weight: sum: 3532.3213  dtype: BFloat16 shape: [3072,8192]
28: model.layers.12.mlp.gate_up_proj.weight: sum: 8034.1665  dtype: BFloat16 shape: [16384,3072]
29: model.layers.12.post_attention_layernorm.weight: sum: 36.9892  dtype: BFloat16 shape: [3072]
30: model.layers.12.self_attn.o_proj.weight: sum: 2360.8115  dtype: BFloat16 shape: [3072,3072]
31: model.layers.12.self_attn.qkv_proj.weight: sum: -39923.85  dtype: BFloat16 shape: [9216,3072]
32: model.layers.13.input_layernorm.weight: sum: 36.9232  dtype: BFloat16 shape: [3072]
33: model.layers.13.mlp.down_proj.weight: sum: 4212.5513  dtype: BFloat16 shape: [3072,8192]
34: model.layers.13.mlp.gate_up_proj.weight: sum: 5825.3145  dtype: BFloat16 shape: [16384,3072]
35: model.layers.13.post_attention_layernorm.weight: sum: 36.8803  dtype: BFloat16 shape: [3072]
36: model.layers.13.self_attn.o_proj.weight: sum: 1412.5374  dtype: BFloat16 shape: [3072,3072]
37: model.layers.13.self_attn.qkv_proj.weight: sum: -1272.3073  dtype: BFloat16 shape: [9216,3072]
38: model.layers.14.input_layernorm.weight: sum: 36.9717  dtype: BFloat16 shape: [3072]
39: model.layers.14.mlp.down_proj.weight: sum: 129.6351  dtype: BFloat16 shape: [3072,8192]
40: model.layers.14.mlp.gate_up_proj.weight: sum: 17178.225  dtype: BFloat16 shape: [16384,3072]
41: model.layers.14.post_attention_layernorm.weight: sum: 36.8547  dtype: BFloat16 shape: [3072]
42: model.layers.14.self_attn.o_proj.weight: sum: -194.2171  dtype: BFloat16 shape: [3072,3072]
43: model.layers.14.self_attn.qkv_proj.weight: sum: 1110.6833  dtype: BFloat16 shape: [9216,3072]
44: model.layers.15.input_layernorm.weight: sum: 36.9053  dtype: BFloat16 shape: [3072]
45: model.layers.15.mlp.down_proj.weight: sum: -3488.173  dtype: BFloat16 shape: [3072,8192]
46: model.layers.15.mlp.gate_up_proj.weight: sum: 3393.5945  dtype: BFloat16 shape: [16384,3072]
47: model.layers.15.post_attention_layernorm.weight: sum: 36.9723  dtype: BFloat16 shape: [3072]
48: model.layers.15.self_attn.o_proj.weight: sum: 2602.2654  dtype: BFloat16 shape: [3072,3072]
49: model.layers.15.self_attn.qkv_proj.weight: sum: 4055.9343  dtype: BFloat16 shape: [9216,3072]
50: model.layers.16.input_layernorm.weight: sum: 36.939  dtype: BFloat16 shape: [3072]
51: model.layers.16.mlp.down_proj.weight: sum: 4393.325  dtype: BFloat16 shape: [3072,8192]
52: model.layers.16.mlp.gate_up_proj.weight: sum: 4148.4727  dtype: BFloat16 shape: [16384,3072]
53: model.layers.16.post_attention_layernorm.weight: sum: 36.9192  dtype: BFloat16 shape: [3072]
54: model.layers.16.self_attn.o_proj.weight: sum: 1939.2538  dtype: BFloat16 shape: [3072,3072]
55: model.layers.16.self_attn.qkv_proj.weight: sum: 3889.2705  dtype: BFloat16 shape: [9216,3072]
56: model.layers.17.input_layernorm.weight: sum: 37.0014  dtype: BFloat16 shape: [3072]
57: model.layers.17.mlp.down_proj.weight: sum: 3038.5884  dtype: BFloat16 shape: [3072,8192]
58: model.layers.17.mlp.gate_up_proj.weight: sum: 3868.7417  dtype: BFloat16 shape: [16384,3072]
59: model.layers.17.post_attention_layernorm.weight: sum: 36.9104  dtype: BFloat16 shape: [3072]
60: model.layers.17.self_attn.o_proj.weight: sum: 4044.3225  dtype: BFloat16 shape: [3072,3072]
61: model.layers.17.self_attn.qkv_proj.weight: sum: 3907.559  dtype: BFloat16 shape: [9216,3072]
62: model.layers.18.input_layernorm.weight: sum: 36.9758  dtype: BFloat16 shape: [3072]
63: model.layers.18.mlp.down_proj.weight: sum: 2598.4373  dtype: BFloat16 shape: [3072,8192]
64: model.layers.18.mlp.gate_up_proj.weight: sum: -1360.5602  dtype: BFloat16 shape: [16384,3072]
65: model.layers.18.post_attention_layernorm.weight: sum: 36.918  dtype: BFloat16 shape: [3072]
66: model.layers.18.self_attn.o_proj.weight: sum: 1755.9471  dtype: BFloat16 shape: [3072,3072]
67: model.layers.18.self_attn.qkv_proj.weight: sum: 3189.6553  dtype: BFloat16 shape: [9216,3072]
68: model.layers.19.input_layernorm.weight: sum: 36.8926  dtype: BFloat16 shape: [3072]
69: model.layers.19.mlp.down_proj.weight: sum: 8881.587  dtype: BFloat16 shape: [3072,8192]
70: model.layers.19.mlp.gate_up_proj.weight: sum: 138151.03  dtype: BFloat16 shape: [16384,3072]
71: model.layers.19.post_attention_layernorm.weight: sum: 37.0612  dtype: BFloat16 shape: [3072]
72: model.layers.19.self_attn.o_proj.weight: sum: 3249.5425  dtype: BFloat16 shape: [3072,3072]
73: model.layers.19.self_attn.qkv_proj.weight: sum: 3827.9995  dtype: BFloat16 shape: [9216,3072]
74: model.layers.2.input_layernorm.weight: sum: 37.0595  dtype: BFloat16 shape: [3072]
75: model.layers.2.mlp.down_proj.weight: sum: 51562.61  dtype: BFloat16 shape: [3072,8192]
76: model.layers.2.mlp.gate_up_proj.weight: sum: 3649.578  dtype: BFloat16 shape: [16384,3072]
77: model.layers.2.post_attention_layernorm.weight: sum: 36.9953  dtype: BFloat16 shape: [3072]
78: model.layers.2.self_attn.o_proj.weight: sum: -10051.838  dtype: BFloat16 shape: [3072,3072]
79: model.layers.2.self_attn.qkv_proj.weight: sum: 3782.4812  dtype: BFloat16 shape: [9216,3072]
80: model.layers.20.input_layernorm.weight: sum: 36.9458  dtype: BFloat16 shape: [3072]
81: model.layers.20.mlp.down_proj.weight: sum: 540.5487  dtype: BFloat16 shape: [3072,8192]
82: model.layers.20.mlp.gate_up_proj.weight: sum: 2291.5325  dtype: BFloat16 shape: [16384,3072]
83: model.layers.20.post_attention_layernorm.weight: sum: 36.8547  dtype: BFloat16 shape: [3072]
84: model.layers.20.self_attn.o_proj.weight: sum: 683.026  dtype: BFloat16 shape: [3072,3072]
85: model.layers.20.self_attn.qkv_proj.weight: sum: 2934.4065  dtype: BFloat16 shape: [9216,3072]
86: model.layers.21.input_layernorm.weight: sum: 36.9986  dtype: BFloat16 shape: [3072]
87: model.layers.21.mlp.down_proj.weight: sum: 3276.8237  dtype: BFloat16 shape: [3072,8192]
88: model.layers.21.mlp.gate_up_proj.weight: sum: -2984.882  dtype: BFloat16 shape: [16384,3072]
89: model.layers.21.post_attention_layernorm.weight: sum: 36.9715  dtype: BFloat16 shape: [3072]
90: model.layers.21.self_attn.o_proj.weight: sum: 1854.0214  dtype: BFloat16 shape: [3072,3072]
91: model.layers.21.self_attn.qkv_proj.weight: sum: 1954.5677  dtype: BFloat16 shape: [9216,3072]
92: model.layers.22.input_layernorm.weight: sum: 36.9588  dtype: BFloat16 shape: [3072]
93: model.layers.22.mlp.down_proj.weight: sum: 3965.9695  dtype: BFloat16 shape: [3072,8192]
94: model.layers.22.mlp.gate_up_proj.weight: sum: 8861.506  dtype: BFloat16 shape: [16384,3072]
95: model.layers.22.post_attention_layernorm.weight: sum: 36.8262  dtype: BFloat16 shape: [3072]
96: model.layers.22.self_attn.o_proj.weight: sum: 2355.2307  dtype: BFloat16 shape: [3072,3072]
97: model.layers.22.self_attn.qkv_proj.weight: sum: 3874.016  dtype: BFloat16 shape: [9216,3072]
98: model.layers.23.input_layernorm.weight: sum: 36.8665  dtype: BFloat16 shape: [3072]
99: model.layers.23.mlp.down_proj.weight: sum: 3408.2212  dtype: BFloat16 shape: [3072,8192]
100: model.layers.23.mlp.gate_up_proj.weight: sum: 7502.153  dtype: BFloat16 shape: [16384,3072]
101: model.layers.23.post_attention_layernorm.weight: sum: 36.8793  dtype: BFloat16 shape: [3072]
102: model.layers.23.self_attn.o_proj.weight: sum: 3519.252  dtype: BFloat16 shape: [3072,3072]
103: model.layers.23.self_attn.qkv_proj.weight: sum: 6876.68  dtype: BFloat16 shape: [9216,3072]
104: model.layers.24.input_layernorm.weight: sum: 36.8804  dtype: BFloat16 shape: [3072]
105: model.layers.24.mlp.down_proj.weight: sum: 2743.601  dtype: BFloat16 shape: [3072,8192]
106: model.layers.24.mlp.gate_up_proj.weight: sum: 4686.205  dtype: BFloat16 shape: [16384,3072]
107: model.layers.24.post_attention_layernorm.weight: sum: 36.8568  dtype: BFloat16 shape: [3072]
108: model.layers.24.self_attn.o_proj.weight: sum: -1500.7341  dtype: BFloat16 shape: [3072,3072]
109: model.layers.24.self_attn.qkv_proj.weight: sum: -422.5952  dtype: BFloat16 shape: [9216,3072]
110: model.layers.25.input_layernorm.weight: sum: 36.9254  dtype: BFloat16 shape: [3072]
111: model.layers.25.mlp.down_proj.weight: sum: 3360.6624  dtype: BFloat16 shape: [3072,8192]
112: model.layers.25.mlp.gate_up_proj.weight: sum: 11327.811  dtype: BFloat16 shape: [16384,3072]
113: model.layers.25.post_attention_layernorm.weight: sum: 36.9898  dtype: BFloat16 shape: [3072]
114: model.layers.25.self_attn.o_proj.weight: sum: 2440.2175  dtype: BFloat16 shape: [3072,3072]
115: model.layers.25.self_attn.qkv_proj.weight: sum: 2857.9844  dtype: BFloat16 shape: [9216,3072]
116: model.layers.26.input_layernorm.weight: sum: 36.9309  dtype: BFloat16 shape: [3072]
117: model.layers.26.mlp.down_proj.weight: sum: 3923.388  dtype: BFloat16 shape: [3072,8192]
118: model.layers.26.mlp.gate_up_proj.weight: sum: 3146.6707  dtype: BFloat16 shape: [16384,3072]
119: model.layers.26.post_attention_layernorm.weight: sum: 36.8458  dtype: BFloat16 shape: [3072]
120: model.layers.26.self_attn.o_proj.weight: sum: 487.4523  dtype: BFloat16 shape: [3072,3072]
121: model.layers.26.self_attn.qkv_proj.weight: sum: -2734.8782  dtype: BFloat16 shape: [9216,3072]
122: model.layers.27.input_layernorm.weight: sum: 36.819  dtype: BFloat16 shape: [3072]
123: model.layers.27.mlp.down_proj.weight: sum: 3937.9905  dtype: BFloat16 shape: [3072,8192]
124: model.layers.27.mlp.gate_up_proj.weight: sum: 4793.846  dtype: BFloat16 shape: [16384,3072]
125: model.layers.27.post_attention_layernorm.weight: sum: 37.0054  dtype: BFloat16 shape: [3072]
126: model.layers.27.self_attn.o_proj.weight: sum: 1370.5074  dtype: BFloat16 shape: [3072,3072]
127: model.layers.27.self_attn.qkv_proj.weight: sum: 1886.34  dtype: BFloat16 shape: [9216,3072]
128: model.layers.28.input_layernorm.weight: sum: 36.9305  dtype: BFloat16 shape: [3072]
129: model.layers.28.mlp.down_proj.weight: sum: 218.0489  dtype: BFloat16 shape: [3072,8192]
130: model.layers.28.mlp.gate_up_proj.weight: sum: 3523.556  dtype: BFloat16 shape: [16384,3072]
131: model.layers.28.post_attention_layernorm.weight: sum: 36.9935  dtype: BFloat16 shape: [3072]
132: model.layers.28.self_attn.o_proj.weight: sum: 2273.84  dtype: BFloat16 shape: [3072,3072]
133: model.layers.28.self_attn.qkv_proj.weight: sum: 5409.1743  dtype: BFloat16 shape: [9216,3072]
134: model.layers.29.input_layernorm.weight: sum: 36.994  dtype: BFloat16 shape: [3072]
135: model.layers.29.mlp.down_proj.weight: sum: 4337.2744  dtype: BFloat16 shape: [3072,8192]
136: model.layers.29.mlp.gate_up_proj.weight: sum: 3414.8083  dtype: BFloat16 shape: [16384,3072]
137: model.layers.29.post_attention_layernorm.weight: sum: 36.9618  dtype: BFloat16 shape: [3072]
138: model.layers.29.self_attn.o_proj.weight: sum: 1339.0393  dtype: BFloat16 shape: [3072,3072]
139: model.layers.29.self_attn.qkv_proj.weight: sum: 4478.0083  dtype: BFloat16 shape: [9216,3072]
140: model.layers.3.input_layernorm.weight: sum: 36.9194  dtype: BFloat16 shape: [3072]
141: model.layers.3.mlp.down_proj.weight: sum: 6475.6895  dtype: BFloat16 shape: [3072,8192]
142: model.layers.3.mlp.gate_up_proj.weight: sum: 2027.214  dtype: BFloat16 shape: [16384,3072]
143: model.layers.3.post_attention_layernorm.weight: sum: 36.9311  dtype: BFloat16 shape: [3072]
144: model.layers.3.self_attn.o_proj.weight: sum: 2732.792  dtype: BFloat16 shape: [3072,3072]
145: model.layers.3.self_attn.qkv_proj.weight: sum: 3951.3384  dtype: BFloat16 shape: [9216,3072]
146: model.layers.30.input_layernorm.weight: sum: 36.8087  dtype: BFloat16 shape: [3072]
147: model.layers.30.mlp.down_proj.weight: sum: 6162.1733  dtype: BFloat16 shape: [3072,8192]
148: model.layers.30.mlp.gate_up_proj.weight: sum: 6202.285  dtype: BFloat16 shape: [16384,3072]
149: model.layers.30.post_attention_layernorm.weight: sum: 36.9597  dtype: BFloat16 shape: [3072]
150: model.layers.30.self_attn.o_proj.weight: sum: 1732.8975  dtype: BFloat16 shape: [3072,3072]
151: model.layers.30.self_attn.qkv_proj.weight: sum: 4454.1406  dtype: BFloat16 shape: [9216,3072]
152: model.layers.31.input_layernorm.weight: sum: 37.0591  dtype: BFloat16 shape: [3072]
153: model.layers.31.mlp.down_proj.weight: sum: 2488.5369  dtype: BFloat16 shape: [3072,8192]
154: model.layers.31.mlp.gate_up_proj.weight: sum: 28962.502  dtype: BFloat16 shape: [16384,3072]
155: model.layers.31.post_attention_layernorm.weight: sum: 36.9939  dtype: BFloat16 shape: [3072]
156: model.layers.31.self_attn.o_proj.weight: sum: 2468.597  dtype: BFloat16 shape: [3072,3072]
157: model.layers.31.self_attn.qkv_proj.weight: sum: 6516.7563  dtype: BFloat16 shape: [9216,3072]
158: model.layers.4.input_layernorm.weight: sum: 36.9538  dtype: BFloat16 shape: [3072]
159: model.layers.4.mlp.down_proj.weight: sum: 4675.616  dtype: BFloat16 shape: [3072,8192]
160: model.layers.4.mlp.gate_up_proj.weight: sum: 3708.774  dtype: BFloat16 shape: [16384,3072]
161: model.layers.4.post_attention_layernorm.weight: sum: 36.9239  dtype: BFloat16 shape: [3072]
162: model.layers.4.self_attn.o_proj.weight: sum: 1671.5455  dtype: BFloat16 shape: [3072,3072]
163: model.layers.4.self_attn.qkv_proj.weight: sum: 5003.91  dtype: BFloat16 shape: [9216,3072]
164: model.layers.5.input_layernorm.weight: sum: 36.8642  dtype: BFloat16 shape: [3072]
165: model.layers.5.mlp.down_proj.weight: sum: 933.2731  dtype: BFloat16 shape: [3072,8192]
166: model.layers.5.mlp.gate_up_proj.weight: sum: 3118.2808  dtype: BFloat16 shape: [16384,3072]
167: model.layers.5.post_attention_layernorm.weight: sum: 36.9298  dtype: BFloat16 shape: [3072]
168: model.layers.5.self_attn.o_proj.weight: sum: 2702.1047  dtype: BFloat16 shape: [3072,3072]
169: model.layers.5.self_attn.qkv_proj.weight: sum: 3267.9353  dtype: BFloat16 shape: [9216,3072]
170: model.layers.6.input_layernorm.weight: sum: 37.0042  dtype: BFloat16 shape: [3072]
171: model.layers.6.mlp.down_proj.weight: sum: 2847.044  dtype: BFloat16 shape: [3072,8192]
172: model.layers.6.mlp.gate_up_proj.weight: sum: -72.5096  dtype: BFloat16 shape: [16384,3072]
173: model.layers.6.post_attention_layernorm.weight: sum: 36.9166  dtype: BFloat16 shape: [3072]
174: model.layers.6.self_attn.o_proj.weight: sum: 745.1311  dtype: BFloat16 shape: [3072,3072]
175: model.layers.6.self_attn.qkv_proj.weight: sum: -1949.0566  dtype: BFloat16 shape: [9216,3072]
176: model.layers.7.input_layernorm.weight: sum: 36.9632  dtype: BFloat16 shape: [3072]
177: model.layers.7.mlp.down_proj.weight: sum: 4007.6667  dtype: BFloat16 shape: [3072,8192]
178: model.layers.7.mlp.gate_up_proj.weight: sum: 3422.3052  dtype: BFloat16 shape: [16384,3072]
179: model.layers.7.post_attention_layernorm.weight: sum: 37.0079  dtype: BFloat16 shape: [3072]
180: model.layers.7.self_attn.o_proj.weight: sum: 2635.744  dtype: BFloat16 shape: [3072,3072]
181: model.layers.7.self_attn.qkv_proj.weight: sum: 2402.4377  dtype: BFloat16 shape: [9216,3072]
182: model.layers.8.input_layernorm.weight: sum: 36.8522  dtype: BFloat16 shape: [3072]
183: model.layers.8.mlp.down_proj.weight: sum: 5040.5347  dtype: BFloat16 shape: [3072,8192]
184: model.layers.8.mlp.gate_up_proj.weight: sum: 3374.0283  dtype: BFloat16 shape: [16384,3072]
185: model.layers.8.post_attention_layernorm.weight: sum: 36.8814  dtype: BFloat16 shape: [3072]
186: model.layers.8.self_attn.o_proj.weight: sum: 2514.216  dtype: BFloat16 shape: [3072,3072]
187: model.layers.8.self_attn.qkv_proj.weight: sum: 3103.429  dtype: BFloat16 shape: [9216,3072]
188: model.layers.9.input_layernorm.weight: sum: 36.944  dtype: BFloat16 shape: [3072]
189: model.layers.9.mlp.down_proj.weight: sum: 32373.86  dtype: BFloat16 shape: [3072,8192]
190: model.layers.9.mlp.gate_up_proj.weight: sum: 3433.8955  dtype: BFloat16 shape: [16384,3072]
191: model.layers.9.post_attention_layernorm.weight: sum: 37.0142  dtype: BFloat16 shape: [3072]
192: model.layers.9.self_attn.o_proj.weight: sum: 900.4572  dtype: BFloat16 shape: [3072,3072]
193: model.layers.9.self_attn.qkv_proj.weight: sum: -2351.5068  dtype: BFloat16 shape: [9216,3072]
194: model.norm.weight: sum: 36.9282  dtype: BFloat16 shape: [3072]
